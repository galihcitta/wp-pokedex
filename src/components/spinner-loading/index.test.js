import React from "react";
import Loading from "./index";
import { StyleSpinner } from "./index.style";

let element;
let spinner;
beforeEach(() => {
    spinner = <Loading.Spinner></Loading.Spinner>;
    element = <Loading isLoading={true}>{spinner}</Loading>;
    element = <Loading isLoading={true}>{spinner}</Loading>;
});

describe("Loading and spinner Component", () => {
    it("renders spinner loading and snapshot element <Loading/>", () => {
        expect(toJson(shallow(element))).toMatchSnapshot();
    });
    it("renders spinner loading props isloading false <Loading/>", () => {
        expect(
            mount(<Loading isLoading={false}></Loading>).prop("isLoading"),
        ).toEqual(false);
    });
    it("renders spinner loading and snapshot element  <spinner/>", () => {
        expect(shallow(spinner).prop("color")).toEqual("aquamarine");
    });
    it("renders spinner loading and snapshot element  <spinner/>", () => {
        expect(
            mountWithTheme(<StyleSpinner color="aquamarine"></StyleSpinner>)
                .find("div")
                .prop("color"),
        ).toEqual("aquamarine");
    });
});
