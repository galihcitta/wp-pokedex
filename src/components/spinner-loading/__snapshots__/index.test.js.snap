// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading and spinner Component renders spinner loading and snapshot element <Loading/> 1`] = `
<styled.div>
  <div
    className="loading-background"
  >
    <div
      className="loading-background-box"
    >
      <Spinner
        color="aquamarine"
      />
    </div>
  </div>
</styled.div>
`;
